### PR TITLE
HOPE Add PDM

### PR DESCRIPTION
**Pull request by pavlo-mk**
[AB#182432](https://unicef.visualstudio.com/ICTD-HCT-MIS/_workitems/edit/182432): move HOPE to PDM

### CODE REVIEW

File: backend/.pdm-python [hunk 1]
```diff
@@ -0,0 +1 @@
+/usr/local/bin/python
```
Comment (wozniakpl) [inline backend/.pdm-python:1]: shouldn't this be gitignored?

**Review by patryk-dabrowski — DISMISSED** [review]
Looks good, could you also update this script https://github.com/unicef/hct-mis/blob/develop/deployment/bump_version.sh ?

**Comment by johniak** [comment]
build is not passing @pavlo-mk

**Comment by johniak** [comment]
@pavlo-mk tests are not passing, fix conflicts

**Comment by johniak** [comment]
@pavlo-mk tests are not passing, fix conflicts

**Comment by johniak** [comment]
@pavlo-mk tests not passing and conflicts

**Comment by pavlo-mk** [comment]
> @pavlo-mk tests not passing and conflicts

yep I saw that, but now working on PR#3788, this one going to be next one.

File: backend/pyproject.toml [hunk 2]
```diff
@@ -75,158 +77,160 @@ disable_error_code = [
...
+    "flaky<4.0.0,>=3.8.1",
+    "pytest-html>=4.1.1",
+]
+
+[tool.pdm.build]
+includes = []
+[project]
 name = "HCT MIS Backend"
-version = "2.4.0"
+version = "2.3.0"
```
Comment (wozniakpl) [inline backend/pyproject.toml:132]: why downgrade?

**Review by patryk-dabrowski — APPROVED** [review]

File: docker/Dockerfile [hunk 3]
```diff
@@ -39,28 +40,34 @@ RUN apt-get update \
...
+    PYTHONPATH=$PYPACKAGES/lib:$CODE:$PYTHONPATH \
+    PATH=$PYPACKAGES/bin:$PATH \
     XDG_RUNTIME_DIR=/run/user/"${UID}"
 
-WORKDIR /code
+WORKDIR $CODE
 
 COPY --from=waitforit /data/waitforit /usr/local/bin/waitforit
 
+RUN pip install pdm==2.15.2 &&\
```
Comment (wozniakpl) [inline docker/Dockerfile:57]: do we need to have pdm in the dist (from base stage)?
can't this be moved to builder stage, as previously?

Comment (pavlo-mk) [reply docker/Dockerfile:57]: sure, good catch

**Review by wozniakpl — APPROVED** [review]

**Review by patryk-dabrowski — APPROVED** [review]

**Pull request merged**